src/pages/orders/index.tsx

```diff
@@ -0,0 +1,159 @@
+import { useState, useCallback, useEffect } from "react";
+import { Heading, InputGroup, Input, InputLeftElement, Flex, Box, Center } from "@chakra-ui/react";
+import { AiOutlineSearch } from "react-icons/ai";
+import { useGetAllOrdersQuery } from "../../redux/api/ordersApi";
+import OrderContainer from "./components/OrderContainer";
+import OrderStatusDropdownFilter from "./components/OrderStatusDropdownFilter";
+import PaginationButton from "@/components/PaginationButton.tsx";
+import { LIMIT, OFFSET } from "@/constants/paginationData";
+import { IOrder } from "@/interfaces/orderData";
+import { FilterObjectType, FilterKey } from "./components/OrderStatusDropdownFilter";
+import _ from "lodash";
+import Error from "@/components/Error/index";
+
+// util for status filtering
+const getFilterQueryString = (filter: FilterObjectType) => {
+  const keys = Object.keys(filter) as FilterKey[];
+  return keys
+    .reduce((acc: string[], key: FilterKey) => {
+      const filterValue = filter[key];
+      return filterValue ? [...acc, `status=${key}`] : acc;
+    }, [])
+    .join("&");
+};
+
+// util for search
+const userIdQueryString = (userId?: string) => {
+  return userId ? `&user_id=${userId}` : "";
+};
```

Comment:
I should create and move it to the Order util folder, so far only have 2, one is the handle checkbox filter, and this one, maybe have more in the future

Reply:
sorry, I still don't think this method is needed. 
1. its parameter is optional. If there is no value, it returns an empty string. What is the point of calling this method? 
2. the backend will also judge whether the _user_id_ is true or false. Regardless of whether _user_id_ is an empty string or undefined, it is false and will not affect the returned data

Reply:
sorry, I still don't think this method is needed. 
1. its parameter is optional. If there is no value, it returns an empty string. What is the point of calling this method? 
2. the backend will also judge whether the _user_id_ is true or false. Regardless of whether _user_id_ is an empty string or undefined, it is false and will not affect the returned data

Reply:
I agree with you, user_id=  and user_id =6372df9c5319b352cdd3f648, the result from the backend is the same.
I wroteif-else util to avoid user_id= appearing at the request's end.
![image](https://user-images.githubusercontent.com/106716226/202851528-fd6ad0f9-5fd2-441f-abe6-54009948454f.png)

Reply:
deleted from the latest commit

Reply:
consider this question again, if no user_id send to backend, shouldn't have 'user_id = ' at the end of the url, even they have the same result